src/Factory/ClientOptions.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageApiBranch\Factory;
+
+use Closure;
+use Keboola\StorageApi\ClientException;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Validator\Constraints\Url;
+use Symfony\Component\Validator\Validation;
+
+class ClientOptions
+{
+    private ?string $url;
+    private ?string $token;
+    private ?string $branchId;
+    private ?string $runId;
+    private ?LoggerInterface $logger;
+    private ?string $userAgent;
+    private ?int $backoffMaxTries;
+    private ?int $awsRetries;
+    private ?bool $awsDebug;
+    private ?Closure $jobPollRetryDelay;
+    private ?Closure $runIdGenerator;
```

Comment:
Ty prvni tri options jsou jasny a jsou hezky videt use-case i v testech. `logger`, `userAgent` a `backoffMaxTries` asi v kontextu API clienta taky chapu, `runId` si jeste taky domyslim.. ale ten zbytek je mi zahadou co to je, proc to je a na co to je?

Reply:
Trošku jsem to doplnil v https://github.com/keboola/storage-api-php-client-branch-wrapper/pull/10 ale k čemu je třeba awsDebug taky nevím :) Dal jsem tam prostě všechny options co je možný nastavit.